src/subscription_manager/managercli.py

```diff
@@ -2675,6 +2677,39 @@ def _do_command(self):
         print("subscription-manager: %s" % self.client_versions["subscription-manager"])
 
 
+class RoleCommand(CliCommand):
+    def __init__(self):
+        shortdesc = _("Modify system purpose role")
+        super(RoleCommand, self).__init__("role", shortdesc, primary=False)
+        self.parser.add_option("--set", dest="set_role",
+                               help=(_("Set role of system purpose")))
+        self.parser.add_option("--unset", dest="unset_role", action="store_true",
+                               help=(_("Unset role of system purpose")))
+
+    def _validate_options(self):
+        if self.options.set_role and self.options.unset_role:
+            system_exit(os.EX_USAGE, _("Error: Options --set and --unset of role subcommand are mutually exclusive."))
+
+    def _do_command(self):
+        self._validate_options()
+        sys_purpurpose_store = SyspurposeStore.read(USER_SYSPURPOSE)
+        if self.options.set_role:
+            sys_purpurpose_store.set('role', self.options.set_role)
+            sys_purpurpose_store.write()
```

Comment:
It would be good to print something to let the user know what happened here like `print(_("System Purpose role has been set to X"))`. Similarly for the unset option below it.